library/src/main/java/org/xmtp/android/library/codecs/ReactionCodec.kt

```diff
@@ -0,0 +1,43 @@
+package org.xmtp.android.library.codecs
+
+import com.google.gson.GsonBuilder
+import com.google.protobuf.kotlin.toByteStringUtf8
+
+val ContentTypeReaction = ContentTypeIdBuilder.builderFromAuthorityId(
+    "xmtp.org",
+    "reaction",
+    versionMajor = 1,
+    versionMinor = 0
+)
+
+data class Reaction(
+    val reference: String,
+    val action: ReactionAction,
+    val content: String,
+    val schema: ReactionSchema,
+)
+
+enum class ReactionAction {
+    ADDED, REMOVED
+}
+
+enum class ReactionSchema {
+    UNICODE, SHORTCODE, CUSTOM
+}
```

Comment:
a brief googling on kotlin enum classes says that these will yield UPPERCASE strings. the TypeScript types for `action` and `schema` expect lowercase values. sorry if my understanding of this is wrong.

Reply:
Yep totally. I just changed them to lowercase.